MaxText/maxengine.py

```diff
@@ -1430,19 +1429,20 @@ def create_engine_from_config_flags(
   """Create new MaxEngine instance with given batch_size, prefill and target lengths, and any config
   params provided through `args_str`.
   """
-  args = {}
-  args["scan_layers"] = "false"
-  args["async_checkpointing"] = "false"
-  args["ici_fsdp_parallelism"] = "1"
-  args["ici_autoregressive_parallelism"] = "1"
-  args["ici_tensor_parallelism"] = "-1"
-  args["weight_dtype"] = "bfloat16"
-  args["attention"] = "dot_product"
+  args = {
+      "scan_layers": "false",
+      "async_checkpointing": "false",
+      "ici_fsdp_parallelism": "1",
+      "ici_autoregressive_parallelism": "1",
+      "ici_tensor_parallelism": "-1",
+      "weight_dtype": "bfloat16",
+      "attention": "dot_product",
+      "max_prefill_predict_length": f"{max_prefill_predict_length}",
+      "max_target_length": f"{max_target_length}",
+      "per_device_batch_size": f"{batch_size}",
+  }
 
   # batch and cache related
```

Comment:
can we move this comment to the list above before max_prefill_predict_length

Reply:
Ok